src/irmin-http/swagger.yml

```diff
@@ -0,0 +1,126 @@
+swagger:  "2.0"
+info:
+  title: "Irmin REST API"
+  description: "This is the first draft of a REST API for the
+    [irmin](https://github.com/mirage/irmin/) database"
+  version: "0.0.1"
+  contact:
+    name: "David Udelson"
+    email: "dru5@cornell.edu"
+paths:
+  /branches:
+    get:
+      summary: "list all branches"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "<response description here>" # TODO
+        404:
+          description: "" #TODO
+    post:
+      summary: "create a new branch"
+      responses:
+        200:
+          description: "" # TODO
+        201:
+          description: "" # TODO
+  /branch/{branchName}:
+    get:
+      summary: "get a branch by name"
+      responses:
+        200:
+          description: "" #TODO
+        404:
+          description: "" #TODO
+    delete:
+      summary: "delete a branch by its ID"
+      responses:
+        200:
+          description: "" #TODO
+        404:
+          description: "" #TODO
+  /commits:
+    get:
+      summary: "lists all commits"
+      responses:
+        200:
+          description: "" #TODO
+        404:
+          description: "" #TODO
+  /commits/{commitHash}:
+    get:
+      summary: "get a commit by its hash"
+      responses:
+        200:
+          description: "" #TODO
+        404:
+          description: "" #TODO
+  /paths:
```

Comment:
I am not sure about `path` here. And the summary doesn't mention `path` at all, so it seems your are not sure either :-) I would recommend adding a short description of what this route is supposed to do and hopefully the right resource name will appear magically!